File: python/백준/그리디/1931 회의실 배정.py
import sys
input = sys.stdin.readline

N = int(input())

data = []

for _ in range(N):
    x, y = map(int, input().split()) 
    data.append((x,y))
  
data.sort(key=lambda x: (x[1], x[0]))

count = 1

end = data[0][1]

for i in range(1, N):
    if data[i][0] >= end:
        end = data[i][1]
        count += 1

print(count) 

File: cpp/백준/그리디/1931 회의실 배정.cpp
#include <iostream>
#include <vector>
#include <algorithm>
using namespace std;

int main() {
    ios_base::sync_with_stdio(false);
    cin.tie(nullptr);

    int N;
    cin >> N;

    vector<pair<int, int>> data(N);
    for (int i = 0; i < N; ++i) {
        int x, y;
        cin >> x >> y;
        data[i] = {x, y};
    }

    sort(data.begin(), data.end(), [](const pair<int, int>& a, const pair<int, int>& b) {
        if (a.second != b.second) return a.second < b.second;
        return a.first < b.first;
    });

    int count = 1;
    int end = data[0].second;

    for (int i = 1; i < N; ++i) {
        if (data[i].first >= end) {
            end = data[i].second;
            count += 1;
        }
    }

    cout << count << "\n";
    return 0;
}
